src/modules/core/api/hooks/useQuery.tsx

```diff
@@ -0,0 +1,44 @@
+import { useEffect, useState, useMemo } from 'react';
+
+export interface Props {
+  url: string;
+  options?: RequestInit;
+}
+
+export interface Return<DT> {
+  data?: DT;
+  error?: Error;
+  loading: boolean;
+}
+
+export default function useQuery<DT>({ url, options }: Props): Return<DT> {
+  const [data, setData] = useState<DT>();
+  const [error, setError] = useState<Error>();
+  const [loading, setLoading] = useState(false);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        setLoading(true);
+        const res = await fetch(url, options);
+        const data = await res.json();
+        setData(data);
+        setLoading(false);
+      } catch (error) {
+        if (error instanceof Error) setError(error);
+        else throw error;
+      }
+    };
+
+    fetchData();
+  }, [url, options]);
+
+  return useMemo(
```

Comment:
what is the point of adding a useMemo here? which cases ae you going to save?

Reply:
For me whenever we have an array or an object I use to use a Memo, in order to memorize the value and not to recalculate each render. It's a little optimization, but in a real application we will hundreds and thousands of arrays/objects, so could represent an efficiency 🤔. This is highly inspired from my last experience where it was like a "rule". I don't know if overusing Memo's and Callback's could also affect the app performance. What your thoughts on that? (interested)